modules/javafx.graphics/src/main/java/com/sun/prism/d3d/D3DShader.java

```diff
@@ -59,19 +59,21 @@ private static native int setConstantsI(long pCtx, long pData, int register,
 
     private static native int nGetRegister(long pCtx, long pData, String name);
 
+    @Override
     public void enable() {
         // res >= 0 is equivalent to D3D's SUCCEEDED(res) macro
         int res = enable(d3dResRecord.getContext().getContextHandle(),
                           d3dResRecord.getResource());
         valid &= res >= 0;
-        d3dResRecord.getContext().validate(res);
+        D3DContext.validate(res);
```

Comment:
This change isn't necessarily equivalent if there are side effects in `D3DRecord::getcontext` that the existing code might be relying upon.

I checked, and there aren't any, so this is just an FYI for future reference.

Reply:
Very interesting, I never considered that actually. So calling a static method on an instance of a class sometimes makes sense?

Reply:
Not really. If `getContext` had side effects, the correct way would be:

```
d3dResRecord.getContext(); // no need to capture the return value 'Context context'
Context.validate(res);
```

The way it's written would be just a shortcut by using the returned instance, but it is not advised to do that anyhow.

Reply:
I would probably characterize that use case as an "unwise" side-effect (at a minimum, a comment would be in order), but it's something to be aware of. Ideally, if that case ever did arise, such that the factory method needed to be called before the static method, I might lean towards making the method in question a "final" instance method rather than a static...